modules/auth/proc-adding-oauth2-proxy-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,83 @@
+[id='proc-adding-oauth2-proxy-as-an-authentication-provider_{context}']
+= Adding OAuth2 proxy as an authentication provider in {product}
+
+The {product} `core-plugin-api` package includes an OAuth2-proxy authentication provider
+that can authenticate users
+by using an link:https://github.com/oauth2-proxy/oauth2-proxy[OAuth2 proxy] in front of an {product} instance.
+This enables to reuse existing authentications within a cluster.
+In general, the OAuth2 proxy supports all OpenID Connect providers.
+For more details, see link:https://oauth2-proxy.github.io/oauth2-proxy/configuration/providers/[the supported provider list].
+
+.Procedure
+
```

Comment:
We should also document how to deploy oauth2-proxy as a sidecar container for RHDH

Reply:
There is a blog post that documents how to do that: https://janus-idp.io/blog/2023/01/17/enabling-keycloak-authentication-in-backstage#deploying-backstage-using-the-backstage-helm-chart


TL;DR for helm chart it requries adding extra container to helm config

```yaml
  extraContainers:
    - name: oauth2-proxy
      env:
        - name: OAUTH2_PROXY_CLIENT_ID
          value: '{{ required "Keycloak Client Secret is Required" .Values.keycloak.clientId }}'
        - name: OAUTH2_PROXY_CLIENT_SECRET
          value: '{{ required "Keycloak Client Secret is Required" .Values.keycloak.clientSecret }}'
        - name: OAUTH2_PROXY_COOKIE_SECRET
          value: '{{ default (randAlpha 32 | lower | b64enc) .Values.keycloak.cookieSecret }}'
        - name: OAUTH2_PROXY_OIDC_ISSUER_URL
          value: '{{ required "Keycloak Issuer URL is Required" .Values.keycloak.issuerUrl }}'
        - name: OAUTH2_PROXY_SSL_INSECURE_SKIP_VERIFY
          value: 'true'
      ports:
        - name: oauth2-proxy
          containerPort: 4180
          protocol: TCP
      imagePullPolicy: IfNotPresent
      image: 'quay.io/oauth2-proxy/oauth2-proxy:latest'
      args:
        - '--provider=oidc'
        - '--email-domain=*'
        - '--upstream=http://localhost:7007'
        - '--http-address=0.0.0.0:4180'
        - '--skip-provider-button'
```

When installing via Operator this is currently not possible to do due to https://issues.redhat.com/browse/RHIDP-1760